contracts/OptimisticArbitrator.sol

```diff
@@ -0,0 +1,159 @@
+ *  confirm or deny the response to a question. To ratify an assertion, an initial assertion must be made and disputed, such that the
+ *  question is escalated to the DVM, where UMA token holders vote on it.
+ * @dev This contract is stateless and solely encapsulates the DVM's functionality to leverage the Optimistic Arbitrator pattern.
+ * @dev The Optimistic Oracle's functions are called on behalf of the caller, hence this contract will neither hold nor receive funds
+ *  from these actions. The Optimistic Oracle in this design is used to handle all bond payouts based on the outcome of the DVM vote.
+ */
+contract OptimisticArbitrator {
+    using SafeERC20 for IERC20;
+
+    FinderInterface public immutable finder;
+
+    OptimisticOracleV2Interface public immutable oo;
+
+    IERC20 public immutable currency;
+
+    bytes32 public priceIdentifier = "YES_OR_NO_QUERY";
+
+    /**
+     * @notice Constructor.
+     * @param _finderAddress finder to use to get addresses of DVM contracts.
+     * @param _currency collateral token used to pay rewards and fees.
+     */
+    constructor(address _finderAddress, address _currency) {
+        finder = FinderInterface(_finderAddress);
+        currency = IERC20(_currency);
+        oo = OptimisticOracleV2Interface(finder.getImplementationAddress(OracleInterfaces.OptimisticOracleV2));
+    }
+
+    /**
+     * @notice Makes an assertion to the Optimistic Oracle.
+     * @dev The Optimistic Oracle price proposal is submitted on behalf of the caller, therefore only the caller,
+     *  and not the Optimist Arbitrator, will hold or receive funds.
+     * @param timestamp timestamp of the price being requested.
+     * @param ancillaryData ancillary data representing additional args being passed with the price request.
+     * @param assertedPrice price being proposed.
+     * @param reward reward offered to a successful proposer. Note: this can be 0, which could make sense if the caller
+     * expects to not being disputed.
+     * @param bond custom proposal bond to set for request. If set to 0, defaults to the final fee.
+     * @param liveness custom proposal liveness to set for request, if set to 0, defaults to the default liveness value.
+     */
+    function makeAssertion(
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        int256 assertedPrice,
+        uint256 reward,
+        uint256 bond,
+        uint64 liveness
+    ) public {
+        _pullAndApprove(reward + bond + _getStore().computeFinalFee(address(currency)).rawValue);
+        _makeAssertion(timestamp, ancillaryData, assertedPrice, reward, bond, liveness);
+    }
+
+    /**
+     * @notice Ratifies an existing assertion to the DVM by disputing the proposed answer to the question.
+     * @dev The proposer and disputer are the same address so the final cost of rafiying an assertion in collateral
```

Comment:
> The proposer and disputer are the same address

The disputer could be different as `ratifyAssertion` can be called by different party from `makeAssertion` caller. This statement applies only for `assertAndRatify`

Reply:
yes, its correct.